solarwinds_apm/sampler.py

```diff
@@ -81,6 +82,26 @@ def __init__(self, apm_config: "SolarWindsApmConfig"):
     def get_description(self) -> str:
         return "SolarWinds custom opentelemetry sampler"
 
+    def construct_url(
+        self,
+        attributes: Attributes = None,
+    ) -> str:
+        """Construct url"""
+        # TODO (NH-34752) Check http scheme, http target, net host name `attributes`
+        #   availability after OTel instrumentation library updates released
+        #   https://github.com/open-telemetry/opentelemetry-python-contrib/issues/936
+        if not attributes:
+            return ""
+        scheme = attributes.get(SpanAttributes.HTTP_SCHEME)
+        host = attributes.get(SpanAttributes.NET_HOST_NAME)
+        port = attributes.get(SpanAttributes.NET_HOST_PORT)
+        target = attributes.get(SpanAttributes.HTTP_TARGET)
+        if scheme and host and target:
+            url = f"{scheme}://{host}:{port}{target}"
```

Comment:
My understanding is that `NET_HOST_PORT` is only required if it doesn't match the scheme (ie 80 for http and 443 for https) so it should probably only be added to the url if present to not end up with urls like `http://google.com:/search`

Reply:
Updated in [acd71d8](https://github.com/solarwindscloud/solarwinds-apm-python/pull/137/commits/acd71d8f4ace9c048b2f3240e6df2746333b8ebd).